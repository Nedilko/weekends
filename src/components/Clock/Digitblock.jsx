import PropTypes from 'prop-types'
import Digit from './Digit'
import Separator from './Separator'

Digitblock.propTypes = {
  label: PropTypes.string.isRequired,
  digit: PropTypes.number.isRequired,
  separator: PropTypes.string,
}

function Digitblock({ label, digit, separator }) {
  return (
    <>
      <div className="flex w-24 flex-col">
        <div className="flex h-8 items-center justify-center text-lg font-thin uppercase">
          {label}
        </div>
        <Digit digit={digit} />
      </div>
      {separator && <Separator symbol={separator} />}
    </>
  )
}

export default Digitblock
