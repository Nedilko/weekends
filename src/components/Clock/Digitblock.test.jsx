import Digitblock from '@components/Clock/Digitblock'
import { render, screen } from '@utils/test-utils'

describe('label and digit are in the document', () => {
  beforeEach(() => {
    render(<Digitblock label="my label" digit={12} />)
  })
  it('label is visible', () => {
    const label = screen.getByText(/my label/i)
    expect(label).toBeInTheDocument()
  })
  it('digit is visible', () => {
    const digit = screen.getByText(/12/i)
    expect(digit).toBeInTheDocument()
  })
})

describe('label, digit and separator are in the document', () => {
  beforeEach(() => {
    render(<Digitblock label="my label" digit={12} separator="--" />)
  })
  it('label is visible', () => {
    const label = screen.getByText(/my label/i)
    expect(label).toBeInTheDocument()
  })
  it('digit is visible', () => {
    const digit = screen.getByText(/12/i)
    expect(digit).toBeInTheDocument()
  })
  it('separator is visible', () => {
    const separator = screen.getByText(/--/i)
    expect(separator).toBeInTheDocument()
  })
})
