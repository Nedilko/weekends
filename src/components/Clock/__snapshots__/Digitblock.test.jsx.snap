// Vitest Snapshot v1

exports[`clock digit block renders > with digit 1`] = `
<div
  class="flex h-20 items-center justify-center text-5xl font-thin"
>
  12
</div>
`;

exports[`clock digit block renders > with label 1`] = `
<div
  class="flex h-8 items-center justify-center text-lg font-thin uppercase"
>
  my label
</div>
`;

exports[`clock digit block renders > with separator 1`] = `
<div
  class="mt-8 flex items-center justify-center pb-2 text-5xl"
>
  :
</div>
`;
